### PR TITLE
Tweak OkHttp client settings for performance

### PR DESCRIPTION
**Pull request by markelliot**
This change is [Reviewable](https://reviewable.io/reviews/palantir/http-remoting/303)

### CODE REVIEW

**Comment by uschi2000** [comment]
:lgtm:

---

Review status: 0 of 3 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/http-remoting/303#-:-KWodKFFsqZzjASs47G8:bnfp4nl)_

**Comment by uschi2000** [comment]
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/http-remoting/303)_

**Review by schlosna — APPROVED** [review]
:+1:

File: jaxrs-clients/src/main/java/com/palantir/remoting1/jaxrs/FeignJaxRsClientBuilder.java [hunk 1]
```diff
@@ -53,12 +54,48 @@
...
 import okhttp3.Credentials;
 import okhttp3.Response;
 import okhttp3.Route;
+import okhttp3.TlsVersion;
 
 public final class FeignJaxRsClientBuilder extends ClientBuilder {
 
+    private static final ImmutableList<ConnectionSpec> CONNECTION_SPEC = ImmutableList.of(
+            new ConnectionSpec.Builder(ConnectionSpec.MODERN_TLS)
+                    .tlsVersions(TlsVersion.TLS_1_2, TlsVersion.TLS_1_1, TlsVersion.TLS_1_0)
```
Comment (schlosna) [inline jaxrs-clients/src/main/java/com/palantir/remoting1/jaxrs/FeignJaxRsClientBuilder.java]: Wait, why do we have to add TLSv1.1 and TLSv1.0 back?

**Comment by markelliot** [comment]
Noted in the commit message: Dropwizard doesn't support anything but TLSv1 and SSLv3.

File: jaxrs-clients/src/main/java/com/palantir/remoting1/jaxrs/FeignJaxRsClientBuilder.java (hunk 1 above)
Comment (markelliot) [reply jaxrs-clients/src/main/java/com/palantir/remoting1/jaxrs/FeignJaxRsClientBuilder.java]: Dropwizard support, still trying to figure out why it's not getting enabled.

**Comment by markelliot** [comment]
@uschi2000 @schlosna this required building with oraclejdk8 -- we still support Java 7 with source compat set to 1.7. I've opened #305 to track removing Java 7 support (IMO worth doing sooner rather than later). Please take one more look at LMK if you're okay with this set of changes.

**Comment by uschi2000** [comment]
👍

**Comment by schlosna** [comment]
Seems sane to me. Worth testing after merge with the one client that needs Java 7 runtime support.

**Pull request merged**